catalog/content/core/content-core-videothumbnailplugin/src/main/java/ddf/content/plugin/video/VideoThumbnailPlugin.java

```diff
@@ -0,0 +1,242 @@
+
+    private static final int MAX_FFMPEG_PROCESSES = 4;
+
+    private final Semaphore limitFFmpegProcessesSemaphore;
+
+    private final String ffmpegPath;
+
+    public VideoThumbnailPlugin(final BundleContext bundleContext) throws IOException {
+        final String ffmpegFolderPath = FilenameUtils.concat(SystemUtils.getUserDir()
+                .getCanonicalPath(), "bin_third_party/ffmpeg");
+
+        if (SystemUtils.IS_OS_LINUX) {
+            ffmpegPath = FilenameUtils.concat(ffmpegFolderPath, "ffmpeg");
+            try (final InputStream inputStream = bundleContext.getBundle()
+                    .getEntry("linux/ffmpeg")
+                    .openStream()) {
+                copyFFmpegBinary(inputStream);
+            }
+        } else if (SystemUtils.IS_OS_MAC) {
+            ffmpegPath = FilenameUtils.concat(ffmpegFolderPath, "ffmpeg");
+            try (final InputStream inputStream = bundleContext.getBundle()
+                    .getEntry("osx/ffmpeg")
+                    .openStream()) {
+                copyFFmpegBinary(inputStream);
+            }
+        } else if (SystemUtils.IS_OS_WINDOWS) {
+            ffmpegPath = FilenameUtils.concat(ffmpegFolderPath, "ffmpeg.exe");
+            try (final InputStream inputStream = bundleContext.getBundle()
+                    .getEntry("windows/ffmpeg.exe")
+                    .openStream()) {
+                copyFFmpegBinary(inputStream);
+            }
+        } else {
+            throw new RuntimeException(
+                    "OS is not Linux, Mac, or Windows. No FFmpeg binary is available for this OS, so the plugin will not work.");
+        }
+
+        limitFFmpegProcessesSemaphore = new Semaphore(MAX_FFMPEG_PROCESSES, true);
+    }
+
+    /**
+     * Deletes the directory that holds the FFmpeg binary.
+     * <p>
+     * Called by Blueprint.
+     */
+    public void destroy() {
+        if (ffmpegPath != null) {
+            FileUtils.deleteQuietly(new File(FilenameUtils.getFullPathNoEndSeparator(ffmpegPath)));
+        }
+    }
+
+    private void copyFFmpegBinary(final InputStream inputStream) throws IOException {
+        final File ffmpegBinary = new File(ffmpegPath);
+
+        if (!ffmpegBinary.exists()) {
+            FileUtils.copyInputStreamToFile(inputStream, ffmpegBinary);
+            if (!ffmpegBinary.setExecutable(true)) {
+                LOGGER.warn(
+                        "Couldn't make FFmpeg binary at {} executable. It must be executable by its owner for the plugin to work.",
+                        ffmpegPath);
+            }
+        }
+    }
+
+    @Override
+    public CreateResponse process(final CreateResponse input) throws PluginExecutionException {
+        // TODO: How to handle application/octet-stream?
+        if (!input.getCreatedContentItem()
+                .getMimeType()
+                .getPrimaryType()
+                .equalsIgnoreCase("video")) {
```

Comment:
perhaps use the guava mediatype constants instead? http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/net/MediaType.html#ANY_VIDEO_TYPE